src/api/app/jobs/report_to_scm_job.rb

```diff
@@ -0,0 +1,27 @@
+# We don't properly capitalize SCM in the class name since CreateJob is doing `CLASS_NAME.to_s.camelize.safe_constantize`
+class ReportToScmJob < CreateJob
+  ALLOWED_EVENTS = ['Event::BuildFail', 'Event::BuildSuccess'].freeze
+
+  queue_as :scm
+
+  def perform(event_id)
+    event = Event::Base.find(event_id)
+    return false unless event.undone_jobs.positive?
```

Comment:
In combination with the comment above changing the finder method.

```suggestion
    return false unless event && event.undone_jobs.positive?
```

Reply:
Following previous comment logic, this change is not needed